addons/web/static/src/js/views/graph/graph_owl_renderer.js

```diff
@@ -0,0 +1,927 @@
+         * The idea is that the getLabel function is in general not invertible but
+         * it is when restricted to the set of dataPoints coming from a same origin.
+
+         * @private
+         * @param {Array} label
+         * @param {Array} originIndex
+         * @returns {string}
+         */
+        _relabelling(label, originIndex) {
+            if (label.isNoData) {
+                return label[0];
+            }
+            var i = this.props.comparisonFieldIndex;
+            if (_.contains(['bar', 'line'], this.props.mode) && i === 0) {
+                // here label is an array of length 1 and contains a number
+                return this.dateClasses.representative(label, originIndex) || '';
+            } else if (this.props.mode === 'pie' && i === 0) {
+                // here label is an array of length at least one containing string or numbers
+                var labelCopy = label.slice(0);
+                if (originIndex !== undefined) {
+                    labelCopy.splice(i, 1, this.dateClasses.representative(label[i], originIndex));
+                } else {
+                    labelCopy.splice(i, 1, this.dateClasses.dateClassMembers(label[i]));
+                }
+                return labelCopy.join('/');
+            }
+            // here label is an array containing strings or numbers.
+            return label.join('/') || _t('Total');
+        }
+
+        /**
+         * Used to format correctly the values in tooltips and yAxes
+         *
+         * @private
+         * @param {number} value
+         * @returns {string} The value formatted using fieldUtils.format.float
+         */
+        _formatValue(value) {
+            //debugger;
+            var measureField = this.fields[this.props.measure];
+            var formatter = fieldUtils.format.float;
+            var formatedValue = formatter(value, measureField, FORMAT_OPTIONS);
+            return formatedValue;
+        }
+
+        /**
+         * Returns the options used to generate the chart legend.
+         *
+         * @private
+         * @param {Number} datasetsCount
+         * @returns {Object}
+         */
+        _getLegendOptions(datasetsCount) {
+            var legendOptions = {
+                display: datasetsCount <= MAX_LEGEND_LENGTH,
+                position: 'top',
+                // TODO
+                //onHover: this._onlegendTooltipHover.bind(this),
+                //onLeave: this._onLegendTootipLeave.bind(this),
+            };
+            var self = this;
+            if (_.contains(['bar', 'line'], this.state.mode)) {
+                var referenceColor;
+                if (this.state.mode === 'bar') {
+                    referenceColor = 'backgroundColor';
+                } else {
+                    referenceColor = 'borderColor';
+                }
+                legendOptions.labels = {
+                    generateLabels: function (chart) {
```

Comment:
Same remark as above, chat with Stefano?

Reply:
As discussed on Discord, not a necessity.